src/printer.js

```diff
@@ -476,10 +486,10 @@ function embed(path, print, textToDoc, opts) {
         // so we remove the last element of the array
         const [formattedStyles, ,] = textToDoc(node.content.styles, { ...opts, parser: parserLang });
 
-        const attributes = path.map((childPath) => childPath.call(print), 'attributes');
-        const styleGroup = group(['<style', indent(group(attributes)), softline, '>']);
-
-        return group([styleGroup, indent([hardline, formattedStyles]), hardline, '</style>', hardline]);
```

Comment:
I just learned that in Prettier, a `group()` is only for elements that could possibly fit on one line. So this may have been messing things up trying to use a group on something that will never fit on one line.